src/HistoryQuery/HistoryQuery.class.js

```diff
@@ -179,15 +193,21 @@ class HistoryQuery {
           this.statusData.scanGroup = `${scanGroup.scanMode} - ${scanGroup.aggregate}`
           this.updateStatusDataStream()
           // eslint-disable-next-line no-await-in-loop
-          await this.exportScanMode(scanMode)
+          const exportResult = await this.exportScanMode(scanMode)
+          if (exportResult === -1) {
+            return
+          }
         } else {
           this.logger.error(`scanMode ${scanMode} ignored: scanGroup.points undefined or empty`)
         }
       }
       this.statusData.scanGroup = null
       this.updateStatusDataStream()
     } else {
-      await this.exportScanMode(this.dataSource.scanMode)
+      const exportResult = await this.exportScanMode(this.dataSource.scanMode)
+      if (exportResult === -1) {
+        return
```

Comment:
should we trace the condition?

Reply:
I didn't because normally it is trace in the exportScanMode method. I can add a logger.trace here though